quint/src/docs.ts

```diff
@@ -1,20 +1,62 @@
+/* ----------------------------------------------------------------------------------
+ * Copyright (c) Informal Systems 2023. All rights reserved.
+ * Licensed under the Apache 2.0.
+ * See License.txt in the project root for license information.
+ * --------------------------------------------------------------------------------- */
+
+/**
+ * Assembling of documentation strings collected from the Quint IR,
+ * previously parsed from documentation comments starting with ///
+ *
+ * @author Gabriela Moreira
+ *
+ * @module
+ */
+
 import { definitionToString } from "./IRprinting";
 import { QuintModule } from "./quintIr";
+import { compact } from 'lodash'
 
+/**
+ * A documentation entry for a definition, compatible with LSP responses for signature help
+ */
 export interface DocumentationEntry {
-  header: string;
-  description?: string;
+  /* The mode and type signature of the definition */
+  label: string;
+  /* The documentation string in markdown for the definition, if present */
+  documentation?: string;
 }
 
-export function produceDocs(quintModule: QuintModule): DocumentationEntry[] {
-  return quintModule.defs.map((def) => {
-    return {
-      header: definitionToString(def, false),
-      description: def.doc,
+/**
+ * Produces a documentation map for the modules' definitions
+ *
+ * @param quintModule the module for which documentation should be produced
+ * @returns a map of definition names to their documentation
+ */
+export function produceDocs(quintModule: QuintModule): Map<string, DocumentationEntry> {
+  const entries = quintModule.defs.map((def) => {
+    if (def.kind === 'module') {
+      // TODO: Produce documentation for nested modules
+      return undefined
```

Comment:
Why return undefined here rather than an empty map?

Reply:
Ah, I think it relates to the use of compact from lodash? A comment explaining would have helped me here.